src/commands/lint/lint.s3.paths.ts

```diff
@@ -0,0 +1,70 @@
+import { command, positional, string } from 'cmd-ts';
+
+import { CliInfo } from '../../cli.info.js';
+import { logger } from '../../log.js';
+import { config, registerCli, verbose } from '../common.js';
+import { imageryBucketNames, imageryCrs, imageryProducts, regions } from './lint.constants.js';
+
+export const commandLintInputs = command({
+  name: 'lint-inputs',
+  description: 'Pretty print JSON files',
+  version: CliInfo.version,
+  args: {
+    config,
+    verbose,
+    path: positional({
+      type: string,
+      displayName: 'path',
+      description: 'JSON file to load inputs from, must be a JSON Array',
+    }),
+  },
+
+  async handler(args) {
+    registerCli(this, args);
+    const startTime = performance.now();
+
+    logger.info('LintInputs:Start');
+    if (!args.path) {
+      throw new Error('No path Provided');
+    }
+    logger.info({ target: args.path }, 'LintInputs:Info');
+    lintPath(args.path);
+    logger.info({ duration: performance.now() - startTime }, 'LintInputs:Done');
+  },
+});
+
+export function lintPath(path: string): void {
+  const pathArray: string[] = path.replace('s3://', '').split('/');
```

Comment:
This can end up splitting the path into more than four components. You probably want to use [the `limit` argument](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/split).

Reply:
I would approach this differently.

you should use `new URL(path)` to validate that the object is a properly formatted url.

then from the url you can gain access to `hostname` and `pathname`

`hostname` is the bucket
`pathname` can then be split into its components, 

Oversplitting the pathname could (should?) be validated to so we dont make too many subfolders.

Reply:
Agreed, using `URL` to parse the URL is probably going to be much more reliable.